### PR TITLE
Add error information to the invoice

### PR DESCRIPTION
**Pull request by majecty**
Next todo:
* Move JSON serialization code to the RPC module.
* Filter meaningless errors before return from RPC

I will do the next todos after JinGyeong came back

### CODE REVIEW

**Comment by kseo** [comment]
@majecty Would you resolve conflicts?

File: types/src/invoice/parcel_invoice.rs [hunk 1]
```diff
@@ -15,36 +15,76 @@
...
+use super::transaction_invoice::TransactionInvoice;
 
-#[derive(Clone, Debug, PartialEq, Serialize)]
-#[serde(untagged)]
+#[derive(Clone, Debug, PartialEq)]
 pub enum ParcelInvoice {
-    Single(Invoice),
-    Multiple(Vec<Invoice>),
+    SingleSuccess,
+    SingleFailed(Error),
```
Comment (kseo) [inline types/src/invoice/parcel_invoice.rs]: Rename `SingleFailed` to `SingleFail`.

File: types/src/parcel/error.rs [hunk 2]
```diff
@@ -78,6 +80,85 @@ pub enum Error {
     InvalidTransaction(TransactionError),
 }
 
+impl Encodable for Error {
+    fn rlp_append(&self, s: &mut RlpStream) {
+        match self {
+            Error::ParcelAlreadyImported => s.begin_list(1).append(&1u8),
```
Comment (kseo) [inline types/src/parcel/error.rs]: Extract numbers as constants.

File: types/src/util.rs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use rlp::{Decodable, DecoderError, Encodable, RlpStream, UntrustedRlp};
+
+use unexpected::Mismatch;
+
+pub fn encode_mismatch<'a, T>(mismatch: &Mismatch<T>, s: &'a mut RlpStream) -> &'a mut RlpStream
```
Comment (kseo) [inline types/src/util.rs]: Move `Mismatch` to `types` package and implement `Encodable` and `Decodable` traits there.

File: types/src/invoice/parcel_invoice.rs [hunk 4]
```diff
@@ -15,36 +15,76 @@
...
 }
 
 impl Encodable for ParcelInvoice {
     fn rlp_append(&self, s: &mut RlpStream) {
         match self {
-            ParcelInvoice::Single(invoice) => {
-                s.append_single_value(invoice);
+            ParcelInvoice::SingleSuccess => {
+                s.begin_list(1);
+                s.append(&1u8);
```
Comment (sgkim126) [inline types/src/invoice/parcel_invoice.rs]: Use a const variable instead of magic value.

**Comment by majecty** [comment]
@sgkim126 I applied your requests.

**Comment by sgkim126** [comment]
LGTM

**Pull request merged**